lib/active_record_host_pool/pool_proxy_6_1.rb

```diff
@@ -39,7 +46,7 @@ def __setobj__(pool_config)
     def connection(*args)
       real_connection = _unproxied_connection(*args)
       _connection_proxy_for(real_connection, @config[:database])
-    rescue Mysql2::Error, ActiveRecord::NoDatabaseError
+    rescue RESCUABLE_DB_ERROR, ActiveRecord::NoDatabaseError, ActiveRecord::StatementInvalid
```

Comment:
How come `ActiveRecord::StatementInvalid` was added?

Reply:
Basically, seems that's just what Trilogy bubbles up:

Trilogy: https://github.com/trilogy-libraries/activerecord-trilogy-adapter/blob/254891b57400a9b35a845d694e53586122a2c114/test/lib/active_record/connection_adapters/trilogy_adapter_test.rb#L291

Mysql2: https://github.com/brianmario/mysql2/blob/6cf5e1d732b4a6ee5485def98637f100cbc86f1b/spec/mysql2/client_spec.rb#L1151

Reply:
👍🏼 I wonder if there are other exceptions which also won’t be wrapped.